enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,513 @@
+
+At the moment, the [Infrastructure](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L14)
+resource serves as a primary source of information about the underlying infrastructure provider and provider-specific parameters.
+Specifically the [PlatformSpec](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L47)
+and [PlatformStatus](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L65) parts
+of the infrastructure resource.
+
+Given that PlatformSpec and PlatformStatus are defined as "discriminated unions" and are required to have the platform type encoded within
+"openshift/api" beforehand, it creates a significant footprint and plenty of effort from Red Hat engineers to create the initial technical enablement of
+a new cloud provider and mostly impossible without Red Hat engineering engagement.
+
+Since a lot of infrastructure related components (such as [CCCMO](https://github.com/openshift/cluster-cloud-controller-manager-operator), [Machine API operator](https://github.com/openshift/machine-api-operator), [Machine Config Operator](https://github.com/openshift/machine-config-operator), and so on)
+require information about, at least, the presence of an underlying cloud provider, the "None" platform does not fit well as a signal in such a situation.
+
+A special, built-in, and somewhat generic platform type that will signal about the presence of an underlying infrastructure platform without
+platform-specific details will help to reduce the number of changes across OCP repositories
+and simplify the initial integration work for non-Red Hat contributors.
+Such a new ability will allow smaller / regional cloud providers to build and test their integration with the OpenShift
+with considerably less effort.
+
+Additionally, there are difficulties that are present today due to having a predefined list of platforms.
+A few examples:
+- No defined mechanism to set `--cloud-provider=external` arg to kubelet/KCM/apiserver without merges and further revendoring of "openshift/api",
+  at the moment decision-making here is tied to the PlatformType.
+- No way to extend machine-api and deliver a new provider without merges to "openshift/machine-api-operator" and "openshift/api" repos.
+
+### Conjunction with Capabilities
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+approach might help to solve the issue and give a room for supplementing *some* platform-dependent components, however some integral parts of
+OpenShift can not be disabled and still requires a signal about an underlying platform, for example, KCM and MCO, in respect of the enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
```

Comment:
Capabilities is a stepping stone before platform operators right? Do we want to talk about this in terms of platform operators rather than capabilities?

Reply:
we could talk more about platform operators here, but i am a little cautious about us over-extending on predictions about the POs. we will need to have more planning with the platform operator effort to ensure that things will work as we'd like. i would be ok with updating this enhancement, or adding a new enhancement, when we understand how we'll use the POs better.

nothing currently in the platform operator description talks about the notion of disabling or enabling core openshift functionality through the addition of a PO. we will need to use the capabilities for now so that we can properly configure the core pieces of the cluster. i think we still have some work to do with respect to the POs to understand these interactions better.